### PR TITLE
Add Account Secret Management Ruby Code Snippets

### PR DESCRIPTION
**Pull request by benhayehudi**
This PR adds Ruby code snippets for secret management

### CODE REVIEW

File: account/secret-management/fetch-a-secret.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'dotenv'
+Dotenv.load
+
+NEXMO_API_KEY = ENV['NEXMO_API_KEY']
+NEXMO_API_SECRET = ENV['NEXMO_API_SECRET']
+
+SECRET_ID = ENV['NEXMO_SECRET_ID']
```
Comment (mheap) [inline account/secret-management/fetch-a-secret.rb]: This should be `NEXMO_SECRET_ID`

File: account/secret-management/revoke-a-secret.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'dotenv'
+Dotenv.load
+
+NEXMO_API_KEY = ENV['NEXMO_API_KEY']
+NEXMO_API_SECRET = ENV['NEXMO_API_SECRET']
+
+SECRET_ID = ENV['NEXMO_SECRET_ID']
```
Comment (mheap) [inline account/secret-management/revoke-a-secret.rb]: This should be `NEXMO_SECRET_ID`

File: account/secret-management/revoke-a-secret.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+require 'nexmo'
+
+client = Nexmo::Client.new(
+    api_key: NEXMO_API_KEY,
+    api_secret: NEXMO_API_SECRET
+)
+
+client.secrets.revoke(SECRET_ID)
```
Comment (mheap) [inline account/secret-management/revoke-a-secret.rb]: We should output `OK` if everything goes well.

Can we also add some error handling for if this fails e.g. the secret ID provided does not exist?

File: account/secret-management/fetch-a-secret.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+require 'nexmo'
+
+client = Nexmo::Client.new(
+    api_key: NEXMO_API_KEY,
+    api_secret: NEXMO_API_SECRET
+)
+
+result = client.secrets.get(SECRET_ID)
+result_hash = result._embedded.secrets[0].to_h
```
Comment (mheap) [inline account/secret-management/fetch-a-secret.rb]: This doesn't look right to me. The response from this endpoint looks like the following (note that there's no `_embedded`:

```
{
  "_links": {
    "self": {
      "href": "/accounts/7c9g88ea/secrets/32399387-1acf-4bb8-af6d-4ae377bde832"
    }
  },
  "id": "32399387-1acf-4bb8-af6d-4ae377bde832",
  "created_at": "2018-07-13T14:40:29Z"
}
```

**Comment by benhayehudi** [comment]
@mheap - Thanks for the review. I've addressed your comments in the latest changes.

**Comment by timcraft** [comment]
Some thoughts looking at the diff...

* You can use `require 'dotenv/load'` to eliminate a couple of lines from each snippet.

* There's no need to convert a symbol to a string to test for equality in the revoke snippet—you can just do `response == :no_content` (instead of calling #to_s).

* The `client.secrets.revoke` method appears to be called twice?

* The whitespace on the list snippet could do with fixing up to be consistent with the other snippets and Ruby norms (2 spaces, newline at end of file).

**Comment by benhayehudi** [comment]
> Some thoughts looking at the diff...
> 

Thanks, @timcraft. That was helpful feedback. I've brought in the changes you mentioned. I also reviewed our current snippets again and I am following the convention set forth in them with:

```
require 'dotenv'

Dotenv.load
```

I'm not sure if we want to introduce a different way of doing it here, and then go back and modify all the existing ones?

**Comment by timcraft** [comment]
> I'm not sure if we want to introduce a different way of doing it here, and then go back and modify all the existing ones?

@benhayehudi Sure, I was looking at it in isolation. Better to be consistent I agree.

**Pull request merged**